extensions/vscode/src/multiStepInputs/newCredential.ts

```diff
@@ -103,6 +188,7 @@ export async function newCredential(
       // when the enableConnectCloud config is turned off
       serverType = ServerType.CONNECT;
       productName = ProductName.CONNECT;
+      resetConnectCloudData(state);
```

Comment:
I think you might want to consider doing a reset sooner in this method, and doing it across all of the different types of data in the "state" at once (required since they are local variables to this module. I'd suggest you keep the independent sub-states, but perhaps just wrap them up into a single collection, which has better management methods.